cisagov/manage.get.gov
# Issues 1584 and 1505: Implement VIP table and fix 401 login bug

## Ticket

Resolves #1584 
Resolves #1505 


## Changes



1584
- Implement VIP table
- Adding an email in django admin to the VIP table records the user who undertook the action
- Emails in the VIP table are exempt from ial2
- Added a reset in auth openid method to avoid 'stuck situations' where a user who's non-vip is logged in session as ial2 and still has to do ial2 even after being added to the VIP table. The reset makes sure everyone starts at ial1 and get bumped to ial2 when right
- Unit tests

1505
- Fixed the 401 bug on login when a user is logged out (and has session of None), the server recompiles, and that user attempts to login again.
- Unit tests




## Context for reviewers



## Setup



### To test the VIP table and auth in general:
- Scenario 1: Login as a new user with a fresh (unauthenticated identitySandbox account). You should be asked to validate your identity/
- Scenario 2: Login as a user who is manager on a domain. You should not be asked to validate.
- Scenario 3: Login as a fresh user who has been invited to a domain. You should not be asked to validate.
- Scenario 4: Login as a fresh user who has been added in DjA to the VIP table. You should not be asked to validate.
- Scenario 5: Login as a fresh user. You will be asked to validate. Add that fresh user to the VIP table in a separate browser. Go back to the browser with the fresh user and type the base URL (getgov-rjm.app.cloud.gov) - you should slide in with ial1 and not be asked to validate.

### To test the 401 fix:
- Login on local
- Logout
- Make a code change that will trigger a recompile
- Login again
- Verify that you login successfully on first try (and you do not get a 401)


## Code Review Verification Steps

### As the original developer, I have

#### Satisfied acceptance criteria and met development standards

- [x] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)


#### Validated user-facing changes (if applicable)


### As a code reviewer, I have

#### Reviewed, tested, and left feedback about the changes

- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)


#### Validated user-facing changes as a developer




**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

### As a designer reviewer, I have

#### Verified that the changes match the design intention


#### Validated user-facing changes as a designer




## Screenshots


![Screen Shot 2024-01-18 at 8 13 43 PM](https://github.com/cisagov/manage.get.gov/assets/107004823/9d8aae55-2b38-45e7-a680-d8711a0750da)
